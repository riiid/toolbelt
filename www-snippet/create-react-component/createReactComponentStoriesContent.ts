import { createImportRaw } from "./createImportRaw.ts";
import { CreateReactComponentOption } from "./CreateReactComponentOption.ts";

export const createReactComponentStoriesContent = (
  name: string,
  option: CreateReactComponentOption,
) => {
  return `import React from "react";
import { Story } from "@storybook/react";
${
    createImportRaw(
      option.onlyNamedExport ? undefined : name,
      option.onlyNamedExport ? [name, `${name}Props`] : [`${name}Props`],
      `./${name}`,
    )
  }

export const Gallery: Story = () => {
  return (
    <${name}></${name}>
  );
};

export const Template: Story<${name}Props> = (props) => {
  return <${name} {...props} />;
};

Template.args = {
  children: "Lorem Ipsum",
};

export default {
  title: "Components/${name}",
  component: ${name},
  parameters: {
    backgrounds: {
      default: "dark",
    },
    layout: "fullscreen",
    viewport: {
      defaultViewport: "galaxys5",
    },
  },
  decorators: [
    (Story: Story) => (
      <Story />
    ),
  ],
};
`;
};
