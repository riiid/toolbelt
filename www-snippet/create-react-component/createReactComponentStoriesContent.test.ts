import { createReactComponentStoriesContent } from "./createReactComponentStoriesContent.ts";
import { assertSnapshot } from "https://deno.land/std@0.138.0/testing/snapshot.ts";

Deno.test(
  `should return correct react component stories.`,
  async (t) => {
    await assertSnapshot(
      t,
      createReactComponentStoriesContent("LoremIpsum", {}),
    );
    await assertSnapshot(
      t,
      createReactComponentStoriesContent("LoremIpsum", {
        onlyNamedExport: true,
      }),
    );
  },
);
