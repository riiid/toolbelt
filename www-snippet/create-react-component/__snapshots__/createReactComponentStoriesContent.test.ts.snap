export const snapshot = {};

snapshot[`should return correct react component stories. 1`] = `
'import React from "react";
import { Story } from "@storybook/react";
import LoremIpsum, { LoremIpsumProps } from "./LoremIpsum";

export const Gallery: Story = () => {
  return (
    <LoremIpsum></LoremIpsum>
  );
};

export const Template: Story<LoremIpsumProps> = (props) => {
  return <LoremIpsum {...props} />;
};

Template.args = {
  children: "Lorem Ipsum",
};

export default {
  title: "Components/LoremIpsum",
  component: LoremIpsum,
  parameters: {
    backgrounds: {
      default: "dark",
    },
    layout: "fullscreen",
    viewport: {
      defaultViewport: "galaxys5",
    },
  },
  decorators: [
    (Story: Story) => (
      <Story />
    ),
  ],
};
'
`;

snapshot[`should return correct react component stories. 2`] = `
'import React from "react";
import { Story } from "@storybook/react";
import { LoremIpsum, LoremIpsumProps } from "./LoremIpsum";

export const Gallery: Story = () => {
  return (
    <LoremIpsum></LoremIpsum>
  );
};

export const Template: Story<LoremIpsumProps> = (props) => {
  return <LoremIpsum {...props} />;
};

Template.args = {
  children: "Lorem Ipsum",
};

export default {
  title: "Components/LoremIpsum",
  component: LoremIpsum,
  parameters: {
    backgrounds: {
      default: "dark",
    },
    layout: "fullscreen",
    viewport: {
      defaultViewport: "galaxys5",
    },
  },
  decorators: [
    (Story: Story) => (
      <Story />
    ),
  ],
};
'
`;
